### ISSUE TITLE
`--write-mode=diff` panics when `TERM=dumb`

### ISSUE THREAD

**Issue by svend** [rust-lang-nursery/rustfmt#1140]
When `TERM=dumb`, `rustfmt --write-mode=diff` panics:

```
$ TERM=dumb rustfmt --write-mode=diff test.rs      
Diff in test.rs at line 29:
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: NotSupported', ../src/libcore/result.rs:788
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

I can get around this by piping through cat:

```
$ TERM=dumb rustfmt --write-mode=diff test.rs | cat
Diff in test.rs at line 29:
         Command::new("sleep")⏎
             .arg("1")⏎
             .status()⏎
-           .unwrap_or_else(|e| panic!("failed to execute process: {}", e));⏎
+            .unwrap_or_else(|e| panic!("failed to execute process: {}", e));⏎
     }⏎
 }⏎
```

**Comment by kamalmarhubi** [comment]
@svend could you give the output of `rustfmt --version`?

**Comment by kamalmarhubi** [comment]
Never mind, fixed in #1142.
